### ISSUE TITLE
marshal: UDT, when values are a map can cause DATA CORRUPTION

### ISSUE THREAD

**Issue by mmatczuk** [gocql/gocql#1624]
### What version of Cassandra are you using?

Any.

### What version of Gocql are you using?

473180d but all versions suffer from it.

### What did you do?

I bound a a map as UDT value, the map did not contain all the UDT values.

Example:

Note the update part where "y" is set to 100

```go
	err := session.ExecStmt(`CREATE KEYSPACE IF NOT EXISTS examples WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}`)
	if err != nil {
		t.Fatal("create keyspace:", err)
	}

	err = session.ExecStmt(`CREATE TYPE IF NOT EXISTS examples.coordinates(x int, y int)`)
	if err != nil {
		t.Fatal("create type:", err)
	}

	err = session.ExecStmt(`CREATE TABLE IF NOT EXISTS examples.udts(k int PRIMARY KEY, c coordinates)`)
	if err != nil {
		t.Fatal("create table:", err)
	}

	coordinates1 := Coordinates{X: 12, Y: 34}
	coordinates2 := Coordinates{X: 56, Y: 78}

	insert := qb.Insert("examples.udts").Columns("k", "c").Query(session)
	insert.BindMap(qb.M{
		"k": 1,
		"c": coordinates1,
	})
	if err := insert.Exec(); err != nil {
		t.Fatal("Exec() failed:", err)
	}
	insert.BindMap(qb.M{
		"k": 2,
		"c": coordinates2,
	})
	if err := insert.Exec(); err != nil {
		t.Fatal("Exec() failed:", err)
	}

	update := qb.Update("examples.udts").Set("c").Where(qb.Eq("k")).Query(session)
	update.Bind(map[string]interface{}{"y": 100}, 1)
	if err := update.Exec(); err != nil {
		t.Fatal("Exec() failed:", err)
	}

	var coordinates []Coordinates
	q := qb.Select("examples.udts").Columns("c").Query(session)
	if err := q.Select(&coordinates); err != nil {
		t.Fatal("Select() failed:", err)
	}

	for _, c := range coordinates {
		t.Logf("%+v", c)
	}
```

This prints 

```
=== RUN   TestIssue2294
    issue_test.go:107: {UDT:<nil> X:100 Y:0}
    issue_test.go:107: {UDT:<nil> X:56 Y:78}
--- PASS: TestIssue2294 (0.05s)
```

This is because the driver [skips the missing values](https://github.com/scylladb/gocql/blob/6aca262598eadfc2c433ac6f790fd8b6abf86f88/marshal.go#L2180) and creates the observed shift in the values.

As defined in the [spec](https://github.com/apache/cassandra/blob/fca2502e1dc05876640396732ec1728e16f1944d/doc/native_protocol_v4.spec#L1003)

> A UDT value is composed of successive [bytes] values, one for each field of the UDT
> value (in the order defined by the type). A UDT value will generally have one value
> for each field of the type it represents, but it is allowed to have less values than
> the type has fields.

TL;DR

User sends the ALL UDT values in the order specified by the UDT.
User may not send the UDT tailing values, the unset values are zeroed.

### What did you expect to see?

An error.

### What did you see instead?

Data corruption.
